backends/audiocd/macosx/macosx-audiocd.cpp

```diff
@@ -188,23 +190,29 @@ void MacOSXAudioCDManager::close() {
 	_trackMap.clear();
 }
 
-enum {
-	// Some crazy high number that we'll never actually hit
-	kMaxDriveCount = 256
-};
-
 MacOSXAudioCDManager::DriveList MacOSXAudioCDManager::detectAllDrives() {
+	int foundDrivesFirst = getfsstat(nullptr, 0, MNT_WAIT);
+	if (foundDrivesFirst <= 0)
+		return DriveList();
+
 	// Fetch the lists of drives
-	struct statfs driveStats[kMaxDriveCount];
-	int foundDrives = getfsstat(driveStats, sizeof(driveStats), MNT_WAIT);
-	if (foundDrives <= 0)
+	struct statfs *driveStats = (struct statfs *)malloc(sizeof(struct statfs) * foundDrivesFirst);
+	int foundDrivesSecond = getfsstat(driveStats, sizeof(struct statfs) * foundDrivesFirst, MNT_NOWAIT);
+	if (foundDrivesSecond <= 0) {
+		free(driveStats);
 		return DriveList();
+	}
+
+	// Can't assume that the values are equal between the two calls; be safe
+	// and only work with the smallest value
+	int foundDrivesMin = MIN(foundDrivesFirst, foundDrivesSecond);
```

Comment:
The way I understand it you can just use the value returned by the second call as I don't think it can be bigger than what it can store in the provided buffer:
> Upon successful completion, the number of statfs structures is returned.
That would mean you don't need two separate variable.

Reply:
Yeah, I just followed the paranoia of that `google/rrg` Rust code I mentioned above, just in case. But indeed I'm not sure why we'd need it.

I dropped it then, let's see how it goes.